pcsx2/GS/Renderers/HW/GSRendererHW.cpp

```diff
@@ -2187,8 +2187,8 @@ void GSRendererHW::Draw()
 
 	m_process_texture = PRIM->TME && !(PRIM->ABE && m_context->ALPHA.IsBlack() && !m_cached_ctx.TEX0.TCC) && !(no_rt && (!m_cached_ctx.TEST.ATE || m_cached_ctx.TEST.ATST <= ATST_ALWAYS));
 
-	const bool no_gaps = PrimitiveCoversWithoutGaps();
-	const bool not_writing_to_all = (!no_gaps || AreAnyPixelsDiscarded() || !all_depth_tests_pass);
+	const NoGapsType no_gaps = PrimitiveCoversWithoutGaps();
```

Comment:
If we're doing this unconditionally here, may as well just get rid of the std::optional (lazy evaluation) and always compute it. Avoid a few branches.

Reply:
Well we still use the optionalness to decide if we've already calculated it to prevent doing it again since this gets called from a few places.

Reply:
Right - I'm meaning just store it as a field in the class, less the optional.

Reply:
Okay I've removed that so it's just a variable now :)